pkg/nfd-master/nfd-master.go

```diff
@@ -440,10 +452,120 @@ func (m *nfdMaster) SetLabels(c context.Context, r *pb.SetLabelsRequest) (*pb.Se
 			klog.Errorf("failed to advertise labels: %v", err)
 			return &pb.SetLabelsReply{}, err
 		}
+
+		// set taints
+		var taints []corev1.Taint
+		if m.args.EnableTaints {
+			taints = crTaints
+		}
+
+		// Call setTaints even though the feature flag is disabled. This
+		// ensures that we delete NFD owned stale taints when flag got
+		// turned off.
+		err = m.setTaints(c, cli, taints, r.NodeName)
+		if err != nil {
+			return &pb.SetLabelsReply{}, err
+		}
 	}
 	return &pb.SetLabelsReply{}, nil
 }
 
+// setTaints sets note taints and annotations based on the taints passed via
+// nodeFeatureRule custom resorce. If empty list of taints is passed, currently
+// NFD owned taints and annotations are removed from the node.
+func (m *nfdMaster) setTaints(c context.Context, cli *kubernetes.Clientset, taints []corev1.Taint, nodeName string) error {
```

Comment:
We haven't been using any context in other parts of the code where interacting with the k8s apiserver (see e.g. `pkg/apihelper/k8shelpers.go`) so I think it would be more consistent to drop it from here as well and use `context.TODO()` when calling `controller.PatchNodeTaints()` below. WDYT?

```suggestion
func (m *nfdMaster) setTaints(cli *kubernetes.Clientset, taints []corev1.Taint, nodeName string) error {
```

Reply:
fine with me, but just a heads up that we have it at least in `SetLabels`.

Reply:
Yeah, that comes from the gRPC server interface but we don't actually use the context for anything 🙄 Maybe we could do that but I think that would probably be better to do in one PR consistently accross all related functions